packages/ia-components/sandbox/details/DetailsDownloadOptions.jsx

```diff
@@ -71,26 +77,31 @@ export default class DetailsDownloadOptions extends React.Component {
             >
               {formats('format', k).downloadable}
               {' '}
-              <I18nIcon className="iconochive-download" en="download"/>
+              <I18nIcon className="iconochive-download" en="download" />
             </AnchorDownload>
+            {(typeof DwebArchive === 'undefined' || !DwebArchive.mirror || (k !== 'Epub')) ? null : (
+              <a href={`/epubreader/index.html?bookPath=/download/${this.props.identifier}/${downloadableFilesDict[k][0].metadata.name}`} title="Read online">
```

Comment:
suggestion: abstract this file URL to a function. this part is my code smell - `downloadableFilesDict[k][0].metadata.name` - what happens when you can't traverse all the way down to get the value? is the link broken?

Reply:
Good question ... but 
- downloadableFilesDict[k] always exists since you are iterating over the keys of downloadableFilesDict. 
- Its always non-empty, because its an inversion, so "k" only exists for members
- files always have metadata which always have a name. 

Sure - abstracted to 
```
function epubReaderUrl(identifier, file) {
    return `/epubreader/index.html?bookPath=/download/${identifier}/${file.metadata.name}`;
  }
```
and
```
href={epubReaderUrl(this.props.identifier, downloadableFilesDict[k][0])}
```